halo2_proofs/src/plonk/circuit.rs

```diff
@@ -1696,15 +1864,18 @@ impl<F: Field> ConstraintSystem<F> {
     ) {
         let mut cells = VirtualCells::new(self);
         let constraints = constraints(&mut cells);
-        let queried_selectors = cells.queried_selectors;
-        let queried_cells = cells.queried_cells;
-
         let (constraint_names, polys): (_, Vec<_>) = constraints
             .into_iter()
             .map(|c| c.into())
-            .map(|c| (c.name, c.poly))
+            .map(|mut c: Constraint<F>| {
+                c.poly.load_to_virtual_cells(&mut cells);
+                (c.name, c.poly)
+            })
             .unzip();
 
+        let queried_selectors = cells.queried_selectors;
+        let queried_cells = cells.queried_cells;
+
```

Comment:
Does this need to move here from its original position?

Reply:
Yes, because `cells.queried_selectors `and `queried_cells` do not got filled in before `load_to_virtual_cells` is called